iModelCore/ecobjects/src/ECSchemaConverter.cpp

```diff
@@ -873,6 +873,10 @@ ECObjectsStatus StandardValuesConverter::ConvertToEnum(ECClassP rootClass, ECCla
         prop->RemoveCustomAttribute(BECA_SCHEMANAME, STANDARDVALUES_CUSTOMATTRIBUTE);
         }
 
+    // Adds a direct reference to the grandparent schema, if missing
+    if (ECObjectsStatus::Success == status)

```

Comment:
This seems misplaced.  There could be an arbitrary depth of hierarchy and multiple branches in the hierarchy.  The algorithm needs to handle adding the reference integrated to setting the type.

Reply:
Moved code to add schema reference in `PrimitiveECProperty::SetType()` function